File: dashboard/frontend/run_tests.js
#!/usr/bin/env node
/**
 * Test runner script for dashboard frontend tests
 */
const { spawn } = require('child_process');
const path = require('path');

function runTests() {
  console.log('🧪 Running Dashboard Frontend Tests...');
  console.log('='.repeat(50));
  
  // Change to frontend directory
  const frontendDir = path.dirname(__filename);
  process.chdir(frontendDir);
  
  return new Promise((resolve, reject) => {
    const testProcess = spawn('npm', ['test', '--', '--watchAll=false', '--verbose'], {
      stdio: 'inherit',
      shell: true
    });
    
    testProcess.on('close', (code) => {
      if (code === 0) {
        console.log('\n✅ All tests passed!');
        resolve(true);
      } else {
        console.log(`\n❌ Tests failed with exit code ${code}`);
        resolve(false);
      }
    });
    
    testProcess.on('error', (error) => {
      console.error(`\n❌ Error running tests: ${error}`);
      reject(error);
    });
  });
}

if (require.main === module) {
  runTests()
    .then(success => process.exit(success ? 0 : 1))
    .catch(error => {
      console.error('Test runner error:', error);
      process.exit(1);
    });
}

module.exports = { runTests };

File: dashboard/backend/run_tests.py
#!/usr/bin/env python3
"""
Test runner script for dashboard backend tests
"""
import subprocess
import sys
import os

def run_tests():
    """Run all backend tests"""
    print("🧪 Running Dashboard Backend Tests...")
    print("=" * 50)
    
    # Change to backend directory
    backend_dir = os.path.dirname(os.path.abspath(__file__))
    os.chdir(backend_dir)
    
    try:
        # Run pytest with verbose output
        result = subprocess.run([
            sys.executable, '-m', 'pytest', 
            'tests/', 
            '-v', 
            '--tb=short',
            '--color=yes'
        ], check=True)
        
        print("\n✅ All tests passed!")
        return True
        
    except subprocess.CalledProcessError as e:
        print(f"\n❌ Tests failed with exit code {e.returncode}")
        return False
    except Exception as e:
        print(f"\n❌ Error running tests: {e}")
        return False

if __name__ == "__main__":
    success = run_tests()
    sys.exit(0 if success else 1)

